libs/content/workspace/src/workspace.rs

```diff
@@ -435,31 +453,92 @@ impl Workspace {
                         }
                     }
                     if sync {
-                        self.perform_sync();
+                        self.tasks.queue_sync();
                     }
                 }
             }
         }
+        if start.elapsed() > Duration::from_millis(100) {
+            warn!("processing completed saves took {:?}", start.elapsed());
+        }
 
+        let start = Instant::now();
         if let Some(sync) = completed_sync {
             self.sync_done(sync)
         }
+        if start.elapsed() > Duration::from_millis(100) {
+            warn!("processing completed sync took {:?}", start.elapsed());
+        }
+
+        let start = Instant::now();
+        if let Some(update) = completed_sync_status_update {
+            self.sync_status_update_done(update)
+        }
+        if start.elapsed() > Duration::from_millis(100) {
+            warn!("processing completed sync status update took {:?}", start.elapsed());
+        }
 
```

Comment:
this amount of timing is making this portion pretty unreadable but your call on whether worthwhile. Maybe just representative of some suffering you incurred.

log messages will all have timestamps and you could always just `instrument` the underlying function. So although the warning helps stick out in the log I would go for overall less log volume (probably the thing that's causing you to add this in the first place).

But actually up to you as the potential debugger of this stuff.

Reply:
What I'm going for here is that when anyone on the team introduces performance issues, the app starts spitting out warnings with a decent indicator of the nature of the issue. Is that different from what you thought? I don't know how log messages having timestamps helps me with that and I'm not sure how to accomplish that by instrumenting a function.

Reply:
yeah I think that is a good idea. There is no way to achieve this trivially, maybe I could add it to our own implementation of a tracing subscriber. 

I just wonder if we've already achieved a good amount more visibility by having the logs in our face in more situations (possibly closing https://github.com/lockbook/lockbook/issues/3108). With a bit of coaching we could prompt team members to look at these in more situations.

Possibly also, for dev-time builds I would be down to timeout lock acquisition as a stronger kick in the butt if the team introduces regressions. 

mainly I'm concerned about the readability hit.

Reply:
feels like we're still pretty early in our journey of logging useful things, is it cool if we do this the messy way until we iron some patterns out, then invest in readability? I am willing to pay the price in readability in this instance to make the logs something I want to have open and watch while I'm developing

maybe a helper like this would go a long way?
```
trait InstantExt {
    fn warn_if_elapsed(&self, duration: Duration)
} 

impl InstantExt for std::time::Instant {
    fn warn_if_elapsed(&self, duration: Duration) {
        if self.elapsed() > duration {
            warn!() // todo maybe this whole thing needs to be a macro so caller can put what they want here
        }
    }
}
```

Reply:
I am down to iron out later

I think this is good use case for that kind of extension trait and helper if you wanted to use it. I think that reduces enough noise to make it a viable pattern immediately

on board with a macro so long as the surrounding block doesn't end up in the macro just the timing statement.

also on board with what you have to just take an `&str` for now, iirc that's all you need at the moment.

but yeah a lil helper in `lb/logging.rs` sounds like a pretty solid addition